### PR TITLE
REDCapTidieR Handling of Non-Repeating Databases

### PR DESCRIPTION
**Pull request by rsh52**
# Description
This PR introduces the argument `has_repeating` which follows the existing `is_longitudinal` detection in the top-level `read_redcap_tidy`. `has_repeating` looks for the presence of `redcap_repeat_instance` in the database output and assigns a value of `TRUE`/`FALSE`.

`has_repeating` then gets fed to `clean_redcap`, `clean_redcap_long`, `extract_nonrepeat_table`, and `extract_nonrepeat_table_long` where it informs conditional execution of repeating instance handling.

To support this I created two new REDCap databases with norepeating instruments, one with arms and one without arms duplicated from existing databases. Their RDS files have been added to this PR.

I updated only the `clean_*` test suites. I think that we can trust if `clean_*` works appropriately then the sub functions are also working as expected.

@skadauke I'm not sure how I like wrapping conditional `if`s around things but wanted to get this addressed and working ASAP once I saw the package was failing for these types of dbs. Lmk if you have any better thoughts on implementation.

# Proposed Changes 
List changes below in bullet format:

- Introduce `has_repeating` to necessary functions
- Update calls to "repeating` logic with conditional execution based on value of `has_repeating`
- Create new RDS file outputs for new nonrepeating test databases
- Update tests for `clean_*` functions to test for proper handling

**Screenshots**
If applicable, add screenshots to help explain the proposed changes

### Issue Addressed
**Closes #[provide issue number to link here]**

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New/revised functions use appropriate naming conventions
- [NA] New/revised functions don't repeat code
   > Code is repeated because of different function requirements
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

**Comment by skadauke** [comment]
Hey, I'm not sure I understand this, let's discuss tomorrow.

**Comment by rsh52** [comment]
> Hey, I'm not sure I understand this, let's discuss tomorrow.

Sure, forgot to include the issue but edited now with #29. The short of it is that our current functions require a check against `redcap_repeat_instance`. These all break when that variable doesn't exist, such as in the case of a database with no repeating instruments. This attempts to address that, right now it works expectedly.

I found this when testing against Will's REDCap sample databases that had no repeating instruments.

**Comment by skadauke** [comment]
I think adding a `has_repeating` argument to pass a flag down through all of these functions has a weird code smell. They are perfectly able to inspect its `db_data` and determine whether there's a `redcap_repeat_instance` column in it and shouldn't rely on another function to tell it this. Why don't you take out the `has_repeating` argument and instead add 

```r
has_repeating <- if("redcap_repeat_instance" %in% names(db_data)) { TRUE } else { FALSE }
```

(or similar) to the beginning of each function that uses it?

File: R/clean_redcap.R [hunk 1]
```diff
@@ -19,44 +19,63 @@ clean_redcap <- function(
...
 
   nonrepeated_forms_tibble <- tibble(
     redcap_form_name = nonrepeated_forms,
     redcap_data = map(
       nonrepeated_forms,
-      ~ extract_nonrepeat_table(.x, db_data, db_metadata)
+      ~ extract_nonrepeat_table(.x,
+                                db_data,
+                                db_metadata,
+                                has_repeating)
```
Comment (skadauke) [inline R/clean_redcap.R:61]: Remove `has_repeating` arg

File: R/clean_redcap.R [hunk 2]
```diff
@@ -86,8 +106,12 @@ extract_nonrepeat_table <- function(
     select(starts_with(my_fields), paste0(my_form, "_complete")) %>%
     names()
```
Comment (skadauke) [inline R/clean_redcap.R:108]: Please modify this function the way you did with `clean_redcap`

File: R/clean_redcap_long.R [hunk 3]
```diff
@@ -74,7 +92,8 @@ extract_nonrepeat_table_long <- function(
     form_name,
     db_data_long,
     db_metadata_long,
-    linked_arms
+    linked_arms,
+    has_repeating
```
Comment (skadauke) [inline R/clean_redcap_long.R:96]: Please modify this function the same way you did with `clean_redcap`

**Review by skadauke — CHANGES_REQUESTED** [review]
Can you also modify the `extract_*_table` functions as indicated in the comments?

**Comment by rsh52** [comment]
@skadauke changes applied, checks passed. This can be merged if you approve and then #33 can be merged as well! 🎉

**Review by skadauke — APPROVED** [review]
Looks good to me!
